website/source/guides/tls.html.md

```diff
@@ -0,0 +1,375 @@
+Nomad's certificate are allowed to access Nomad.
+
+~> Enabling `verify_https_client` feature effectively protects Nomad from
+   unauthorized network access at the cost of breaking compatibility with Consul
+   HTTPS health checks.
+
+### Client configuration
+
+The Nomad client configuration is similar with the only difference being the
+certificate and key used:
+
+```hcl
+# Increase log verbosity
+log_level = "DEBUG"
+
+# Setup data dir
+data_dir = "/tmp/client1"
+
+# Enable the client
+client {
+    enabled = true
+
+    # For demo assume we are talking to server1. For production,
+    # this should be like "nomad.service.consul:4647" and a system
+    # like Consul used for service discovery.
+    servers = ["127.0.0.1:4647"]
+}
+
+# Modify our port to avoid a collision with server1
+ports {
+    http = 5656
+}
+
+# Require TLS
+tls {
+  http = true
+  rpc  = true
+
+  ca_file   = "nomad-ca.pem"
+  cert_file = "client.pem"
+  key_file  = "client-key.pem"
+
+  verify_server_hostname = true
+  verify_https_client    = true
+}
+```
+
+### Running with TLS
+
+Now that we have certificates generated and configuration for a client and
+server we can test our TLS-enabled cluster!
+
+In separate terminals start a server and client agent:
+
+```shell
+# In one terminal...
+nomad agent -config server1.hcl
+
+# ...and in another
+nomad agent -config client1.hcl
+```
+
+Finally in a third terminal test out `nomad node-status`:
+
+```text
+vagrant@nomad:~$ nomad node-status
+Error querying node status: Get http://127.0.0.1:4646/v1/nodes: malformed HTTP response "\x15\x03\x01\x00\x02\x02"
+```
+
+Oh no! That didn't work!
+
+Don't worry, the Nomad CLI just defaults to `http://...` instead of
+`https://...`. We can override this with an environment variable:
+
+```shell
+export NOMAD_ADDR=https://localhost:4646
+export NOMAD_CACERT=nomad-ca.pem
+export NOMAD_CLIENT_CERT=client.pem
+export NOMAD_CLIENT_KEY=client-key.pem
+```
+
+The `NOMAD_CACERT` also needs to be set so the CLI can verify it's talking to
+an actual Nomad node. Finally, `NOMAD_CLIENT_CERT` and `NOMAD_CLIENT_KEY` need
+to be set since we enabled `verify_https_client` above which prevents any
+access lacking a client certificate.
+
+Now the CLI works as expected:
+
+```text
+vagrant@nomad:~$ nomad node-status
+ID        DC   Name   Class   Drain  Status
+237cd4c5  dc1  nomad  <none>  false  ready
+
+vagrant@nomad:~$ nomad init
+Example job file written to example.nomad
+vagrant@nomad:~$ nomad run example.nomad
+==> Monitoring evaluation "e9970e1d"
+    Evaluation triggered by job "example"
+    Allocation "a1f6c3e7" created: node "237cd4c5", group "cache"
+    Evaluation within deployment: "080460ce"
+    Evaluation status changed: "pending" -> "complete"
+==> Evaluation "e9970e1d" finished with status "complete"
+```
+
+## Server Gossip
+
+We haven't quite completely secured Nomad's communications: Nomad server's
```

Comment:
Added more content around the various protocols and can toss in any graphs we dig up.